lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -144,14 +162,41 @@ class DecomposeAtenSoftmaxIntOp : public OpRewritePattern<AtenSoftmaxIntOp> {
     if (!tensorType.hasDtype() || !tensorType.getDtype().isa<mlir::FloatType>())
       return rewriter.notifyMatchFailure(op, "Only support floating type");
 
-    // exp(x)
-    Value exp = rewriter.create<AtenExpOp>(loc, tensorType, self);
-    // sum(exp(x))
-    Value sum = createSumAlongDimension(rewriter, loc, op, exp, dim, /*keepDim=*/true);
-    if (!sum)
+    Value result = getSoftmaxResult(op, tensorType, rewriter);
+    if (!result)
+      return failure();
+    rewriter.replaceOpWithNewOp<TensorStaticInfoCastOp>(op, op.getType(),
+                                                        result);
+    return success();
+  }
+};
+} // namespace
+
+namespace {
+class DecomposeAten_SoftmaxOp : public OpRewritePattern<Aten_SoftmaxOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(Aten_SoftmaxOp op,
+                                PatternRewriter &rewriter) const override {
+    Value self = op.self();
+    BaseTensorType tensorType = self.getType().cast<BaseTensorType>();
+    if (!tensorType.hasDtype() || !tensorType.getDtype().isa<mlir::FloatType>())
+      return rewriter.notifyMatchFailure(op, "Only support floating type");
+    bool halfToFloat;
+    Type finalType;
+    if (matchPattern(op.half_to_float(), m_TorchConstantBool(&halfToFloat))) {
+      finalType = (halfToFloat)
+                      ? tensorType.getWithSizesAndDtype(tensorType.getSizes(),
```

Comment:
The half-type handling is a bit tricky. In particular, it is not clear to me that all that needs to be done is set the `finalType` to `f32`. This might end up causing type mismatches down the line when the `getSoftmaxResult` is calculated because the input tensors will be of a different type compared to the final result. What makes this even more tricky is that we cannot test this on CPU because the `halfToFloat` flag is not supported on CPU.

It may be safer to make the `halfToFloat = true` a match failure saying that it is currently not supported until we can figure out a way to have a unit test for this to guarantee correctness.

Reply:
Added match failure for `halfToFloat = true`